nh_clinical/activity_extension.py

```diff
@@ -538,110 +530,241 @@ def get_activity_user_ids(self, cr, uid, activity_id, context=None):
+        context (Normally eobs)
+
+        :param activity: Policy activity definition
+        :type activity: dict
+        :param location_id: ID of location to check policy for
+        :type location_id: int
+        :return: True if can create activity for location
+        :rtype: bool
+        """
+        location_model = self.env['nh.clinical.location']
+        triggered_context = activity.get('context')
+        if triggered_context and location_id:
+            location = location_model.browse(location_id)
+            # Create a list of bools representing if the activity and location
+            # contexts match. If none of them match then return False
+            if not any(
+                    [c.name == triggered_context for c in
+                     location.context_ids]):
+                return False
+        return True
+
+    @api.model
+    def check_trigger_domains(self, activity, domains=None):
+        """
+        Check if any records are returned for the supplied domains. If so then
+        return False
+
+        :param activity: nh.clinical.activity instance
+        :param domains: list of dicts with object (model to query) and domain
+            (search query to carry out on model)
+        :return: True if no records found, False is records found
+        :rtype: bool
+        """
+        if not domains:
+            return False
+        for domain in domains:
+            domain_pool = self.env[domain['object']]
+            search_domain = domain['domain'] + [
+                ['parent_id', '=', activity.spell_activity_id.id]
+            ]
+            if domain_pool.search(search_domain):
+                return True
+        return False
+
+    @api.model
+    def trigger_policy_cancel_others(self, spell_activity_id, trigger_model):
+        """
+        Cancel all open activities when triggering the policy
+
+        :param spell_activity_id: ID of the spell's activity record. This is
+            used by the cancel_
+        """
+        activity_model = self.env['nh.activity']
+        model_data = self.env['ir.model.data']
+        # Create a dict of the model names and cancel reasons so can do a
+        # lookup while also leaving room to extend with other models
+        reasons = {
+            'nh.clinical.patient.placement': model_data.get_object(
+                'nh_clinical', 'cancel_reason_placement').id
+        }
+        activity_model.cancel_open_activities(
+            spell_activity_id,
+            trigger_model,
+            cancel_reason_id=reasons.get(self._name)
+        )
+
+    def _get_policy_create_data(self, case=None):
+        """
+        Get the create_data dictionary for the policy
+
+        :param case: Case that controls the logic used when getting policy
+        :type case: int
+        :return: Dictionary of values to use when creating a new activity
+        :rtype: dict
+        """
+        return {}
```

Comment:
Would this actually work if it was ever called? Otherwise may be better to raise a `NotImplementedError`.

Reply:
There are only a few subclasses that override this method. These are in `nh_eobs/policy.py`, the reasoning behind this was to have a template method that would define the default behaviour (in this case don't add any new key, value pairs) and then have the subclasses add their own.